### PR TITLE
1165 error on create existing

### LINKED ISSUE #1165: Enhancement: error when resource exists in real infrastructure upon creation
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v1.6.6 on darwin_amd64

### APIC version and APIC Platform

* V 2.13.2 and all.

### Affected Resource(s)

* `aci_user_security_domain`

### Terraform Configuration Files

```hcl
resource "aci_user_security_domain" "example1" {
  local_user_dn  = module.local_fgh001.id
  name  = "example"
  annotation = "orchestrator:terraform"
  name_alias = "example_name_alias"
  description = "from Terraform"
}

resource "aci_user_security_domain" "example2" {
  local_user_dn  = module.local_fgh001.id
  name  = "example"
  annotation = "orchestrator:terraform"
  name_alias = "example_name_alias"
  description = "from Terraform"
}
```

### Debug Output

N/A

### Panic Output

N/A

### Expected Behavior

Terraform should prevent the creation of two `aci_user_security_domain` resources with identical attributes (specifically name and local_user_dn) or properly handle the conflict by notifying the user that a resource with these parameters already exists.

It should also not end up in a scenario, where the resources tracked by the provider creates a scenario here, removing one also deletes the other one.

The Terraform Cisco ACI and/or provider should either:

* Prevent creation of duplicate `aci_user_security_domain` resources with the same name and local_user_dn, or
* Properly handle the scenario, where ACI API's response when attempting to create a duplicate resource, ideally providing a meaningful error message.

### Actual Behavior

Terraform plans and attempts to create two `aci_user_security_domain` resources with the same name and local_user_dn, leading to an ambiguous state where it seems both are created, but actually, only one exists in ACI. Subsequent Terraform runs show inconsistent behavior regarding the creation and destruction of these resources.

See the tracked ID's here

```txt
aci_user_security_domain.example1: Creation complete after 1s [id=uni/userext/user-fgh001/userdomain-example]
aci_user_security_domain.example2: Creation complete after 1s [id=uni/userext/user-fgh001/userdomain-example]
```

### Steps to Reproduce

1. Define two `aci_user_security_domain` resources in your Terraform configuration with identical name and `local_user_dn` attributes.
2. Run `terraform apply` to apply the configuration. Notice that Terraform will attempt to create both resources without indicating any conflict.
3. Observe that only one user security domain is actually created in ACI, but Terraform believes both exist.
4. Remove or comment out one of the `aci_user_security_domain` resource definitions and re-run `terraform apply`. Terraform will now attempt to delete the supposedly existing resource, leading to inconsistencies.
5. Re-add the previously removed `aci_user_security_domain` resource to the Terraform configuration and run terraform apply again. Terraform will attempt to create the resource anew, despite it supposedly having existed before.

### Important Factoids

The behavior suggests a gap in the solutions handling of resource uniqueness and idempotency, particularly in how it interacts with the ACI API to check for existing resources before attempting to create new ones.

### References

* #0000

### PR DESCRIPTION
**Pull request by akinross**
fixes #1165 

requires #1223 to be merged first and then needs rebase

### CODE REVIEW

**Review by samiib — CHANGES_REQUESTED** [review]

File: internal/provider/utils.go [hunk 1]
```diff
@@ -70,7 +68,7 @@ func DoRestRequestEscapeHtml(ctx context.Context, diags *diag.Diagnostics, clien
 		if errCode != "1" && errCode != "103" && errCode != "107" && errCode != "120" {
 			diags.AddError(
 				fmt.Sprintf("The %s rest request failed", strings.ToLower(method)),
-				fmt.Sprintf("Code: %d Response: %s, err: %s. Please report this issue to the provider developers.", restResponse.StatusCode, cont.Data().(map[string]interface{})["imdata"], err),
```
[the remark this answers is not in the document]
Comment (akinross) [reply internal/provider/utils.go:73]: No I am not suggesting that people should not raise issues anymore. 

I think this line should not have been there in the first place and is perceived as confusing. The issue lies that wrong configuration, or any other valid apic response that has nothing to do with the provider could be interpreted as raise issue with the provider. I have had people reaching out to me to report that a redirect policy was already associated with different l3outs, which does not seem like something we as provider developers should be busy with. 

Please let me know if you have a better text to put in place then I can put that there.

File: internal/provider/utils.go [hunk 2]
```diff
@@ -30,14 +30,12 @@ func GetMOName(dn string) string {
...
-	if err != nil {
-		tflog.Error(ctx, fmt.Sprintf("failed validate relation dn: %s", dn))
+func CheckDn(ctx context.Context, diags *diag.Diagnostics, client *client.Client, classname, dn string) {
+	requestData := DoRestRequest(ctx, diags, client, fmt.Sprintf("api/mo/%s.json", dn), "GET", nil)
+	if requestData.Search("imdata").Search(classname).Data() != nil {
 		diags.AddError(
-			"Relation target dn validation failed",
-			fmt.Sprintf("The relation target dn is not found: %s", dn),
+			"object already exists",
+			fmt.Sprintf("The %s object with DN '%s' already exists", classname, dn),
```
[the remark this answers is not in the document]
Comment (akinross) [reply internal/provider/utils.go]: there were more, changed those also

File: internal/provider/utils.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (samiib) [reply internal/provider/utils.go:73]: This makes sense. I thought it was something like this. Basically the message suggested it was a bug in the provider if anything goes wrong even for user config issues.

**Review by sajagana — APPROVED** [review]
LGTM!

**Review by samiib — APPROVED** [review]
LGTM

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**